### PR TITLE
[WIP] Verify that your macOS system has the latest EFI updates installed

### PR DESCRIPTION
**Pull request by alessandrogario**
This PR adds support for verifying the state of your EFI updates, taking into account the hardware model and version. This functionality is provided by querying the Duo Labs EFIgy API (https://efigy.io).

## Additional reading
* https://duo.com/blog/the-apple-of-your-efi-mac-firmware-security-research
* https://duo.com/assets/ebooks/Duo-Labs-The-Apple-of-Your-EFI.pdf

### CODE REVIEW

File: osquery/tables/system/darwin/efigy.cpp [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+      .openssl_sni_hostname("api.efigy.io");
+
+  client_options.timeout(5).follow_redirects(true);
+
+  std::string raw_server_response;
+
+  try {
+    osquery::http::Request server_request("https://api.efigy.io/apple/oneshot");
+
+    server_request << osquery::http::Request::Header("User-Agent", "osquery");
```
Comment (reedloden) [inline osquery/tables/system/darwin/efigy.cpp]: I'd include the version number here.

File: osquery/tables/system/darwin/efigy.cpp [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+#include <osquery/sql.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/remote/http_client.h"
+#include "osquery/tables/system/hash.h"
+
+namespace pt = boost::property_tree;
+
+namespace {
```
Comment (obelisk) [inline osquery/tables/system/darwin/efigy.cpp]: Can we use `namespace osquery` so we don't have `osquery::` for all the internal datatypes?

File: osquery/tables/system/darwin/efigy.cpp [hunk 3]
```diff
@@ -0,0 +1,496 @@
...
+  }
+
+  return osquery::Status(0, "OK");
+}
+
+osquery::Status getOSVersion(std::string& version, std::string& build) {
+  version.clear();
+  build.clear();
+
+  auto sw_vers = osquery::SQL::selectAllFrom(
```
Comment (groob) [inline osquery/tables/system/darwin/efigy.cpp]: why not use the `os_version` table?

File: osquery/tables/system/darwin/efigy.cpp [hunk 4]
```diff
@@ -0,0 +1,496 @@
...
+
+struct CFStringRefDeleter final {
+  using pointer = CFStringRef;
+
+  void operator()(pointer p) {
+    CFRelease(p);
+  }
+};
+
+osquery::Status getBoardID(std::string& board_id) {
```
Comment (groob) [inline osquery/tables/system/darwin/efigy.cpp]: `board_id` would be great to add to the existing `platform_info` table or in a separate table.

**Comment by alessandrogario** [comment]
Implemented as an extension here: https://github.com/trailofbits/osquery-extensions

**Pull request closed without merging**